addons/sale/models/product_template.py

```diff
@@ -232,7 +232,7 @@ def _get_saleable_tracking_types(self):
 
     def _get_product_accounts(self):
         product_accounts = super()._get_product_accounts()
-        product_accounts['downpayment'] = self.categ_id.property_account_downpayment_categ_id
+        product_accounts['downpayment'] = self.company_id.property_downpayment_account_id
```

Comment:
Is it still used somewhere?
Since you remove `_get_down_payment_account` from the invoicing wizard, I don't know why it's still useful here (and potentially wrong if the product is shared between companies)